pennylane/gradients/spsa_gradient.py

```diff
@@ -71,7 +72,7 @@ def spsa_grad(
     shots=None,
     num_directions=1,
     sampler=_rademacher_sampler,
-    sampler_seed=None,
+    sampler_rng=None,
```

Comment:
Do we need to rename this keyword argument, as that would be considered a breaking change. Alternatively, we could just expand the definition of  `sampler_seed` to include generators or generally anything `default_rng` accepts.

The big change is in the call signature for the `sampler`.  That's what actually needs to be accepting a generator instead of a seed.

Reply:
I was also thinking about sticking with the name `sampler_seed`, but I feel like we should encourage users to pass a generator, since they will most likely call `spsa_grad` many times as part of some optimization process. Encouraging them to pass a generator instead of a seed, will probably prevent some users from making the mistake of passing the same seed in each `spsa_grad` call.

But I am also OK with keeping `sampler_seed` and emphasizing the advantage of generators in the documentation.